// @utils/ComVerify.ts
import isColor from "is-color";
import { isUrl } from "./is";

class ComV { }
class UserV {
    static name_(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 15], '用户名字');
    }
    static account(_str: string = ''): string {
        return ComVerify.lengthV(_str, [3, 18], '用户账户');
    }
    static password(_str: string = ''): string {
        return ComVerify.lengthV(_str, [6, 20], '用户密码');
    }
    static state(_str: string = ''): string {
        return ComVerify.lengthV(_str, [0, 100], '用户状态');
    }
    static introduction(_str: string = ''): string {
        return ComVerify.lengthV(_str, [0, 200], '个人简介');
    }
    static memoData(_str: string = ''): string {
        return '';
    }
    static socialData(_str: string = ''): string {
        return '';
    }
    static themeData(_str: string = ''): string {
        return '';
    }
}
class BlogV {
    static title(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 200], '博客标题');
    }
    static content(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 1000 * 1024], '博客内容');
    }
    static tabIds(_tabs: number[]) {
        return '';
    }
    static externalLink(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 300], '链接') || ComVerify.urlV(_str);
    }
}
class BlogTabV {
    static name_(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 10], '标签名字');
    }
    static color(_str: string = ''): string {
        return ComVerify.colorV(_str, '标签颜色');
    }
}
class MemoV {
    static title(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 100], '便签标题');
    }
    static content(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 10000], '便签内容');
    }
}
class DiaryV {
    static title(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 100], '日记标题');
    }
    static content(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 10000], '日记内容');
    }
}
class BulletCommentV {
    static color(_str: string = ''): string {
        return ComVerify.colorV(_str, '弹幕颜色');
    }
    static content(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 100], '弹幕内容');
    }
}
class BlogCommentV {
    static content(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 3000], '评论内容');
    }
    static name_(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 50], '评论名字');
    }
    static img(_str: string = ''): string {
        return '';
    }
}

class DiarySupplementV {
    static content(_str: string = ''): string {
        return ComVerify.lengthV(_str, [1, 3000], '补充内容');
    }
}

/**
 * 公共验证
 */
export class ComVerify {
    /** 公共验证 */
    static ComV = ComV;
    /** 用户验证 */
    static UserV = UserV;
    /** 博客验证 */
    static BlogV = BlogV;
    /** 博客标签验证 */
    static BlogTabV = BlogTabV;
    /** 便签验证 */
    static MemoV = MemoV;
    /** 日记验证 */
    static DiaryV = DiaryV;
    /** 弹幕验证 */
    static BulletCommentV = BulletCommentV;
    /** 博客评论验证 */
    static BlogCommentV = BlogCommentV;
    /** 日记补充验证 */
    static DiarySupplementV = DiarySupplementV;

    /**
     * 颜色验证
     * @param _str 
     * @param _title 
     * @returns 
     */
    static colorV(_str: string, _title: string) {
        return isColor(_str) ? '' : `${_title}必须是正确的颜色值`;
    }

    /**
     * 长度验证
     * @param _str 
     * @param _g 
     * @param _title 
     */
    static lengthV(_str: string, _g: [number, number], _title: string): string {
        if (_str.length < _g[0]) {
            if (_g[0] == 1) {
                return `${_title}不能为空`;
            }
            return `${_title}长度不能少于${_g[0]}个字符`;
        }
        if (_str.length > _g[1]) {
            return `${_title}长度不能超过${_g[1]}个字符`;
        }
        return '';
    }

    /**
     * 
     * @param _str 路径验证
     */
    static urlV(_str: string): string {
        if (isUrl(_str)) { return ''; }
        return '路径格式有误';
    }
}
